calcom/cal.com
# feat: cal.ai v1.2.0

- [x] clean up, comment, more tests
- [x] improve onboarding email copy
- [x] make multiple booking links smaller (currently passes full date, time, timezone but `4pm` would be cleaner
- [x] streamline prompts with simpler instructions + less tokens
- [ ] loom video

## What does this PR do?



### v1.2.0 includes new features:

*Improved onboarding experience*
- Upon install, you get an onboarding email from your AI
- If you didn't install the app and send an email to cal.ai, you get a notice to install with a link to the App Store

*Multiple booking options in the booking link flow*
- Here are some suggested times: [2pm] [3pm] [see all times]

*Better understanding of nuanced flow and suggested actions*
- When booking a meeting, cal.ai should suggest replying if you want to reschedule
- If referencing an external user without the `@username` convention, cal.ai should ask for their `@username` or `email`
- Copy from the AI should be more friendly and simple

*Prevent context loss*
- Email clients (apple, google, etc.) don't pass the full thread, so we encourage the AI to summarize relevant context in each reply. This should help prevent context loss

### Additionally, v1.2 includes chores and bug fixes:

Increase timeout threshold on intelligence route
- [Vercel changed how timeouts work](https://vercel.com/changelog/serverless-functions-can-now-run-up-to-5-minutes) in serverless functions. We need to allow the intelligence route more time to cook.

Bump nextjs version 
- The maxDuration feature implemented above ^ requires a bump to nextjs version. Could go with 13.4.7 or go all the way to 13.5.4 🤷‍♂️



LOOM COMING SOON

## Requirement/Documentation



- ~If there is a requirement document, please, share it here.~
- ~If there is ab UI/UX design document, please, share it here.~

## Type of change




## How should this be tested?



To test and deploy, first set new env vars:

In the AI app we now need: 
- ~`CALAI_API_KEY` -> should just be a long random string~
- `SENDER_DOMAIN` -> should be `cal.ai` in prod

In the App Store Env we now need:
- ~`CALAI_API_KEY` -> should match the AI env~
- ~`CALAI_URL` -> should be `https://cal.ai` in prod, `http://localhost:3005` in local~

## Mandatory Tasks


## Checklist



- ~I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)~
- ~My code doesn't follow the style guidelines of this project~
- ~I haven't commented my code, particularly in hard-to-understand areas~
- ~I haven't checked if my PR needs changes to the documentation~
- ~I haven't checked if my changes generate no new warnings~
- I haven't added tests that prove my fix is effective or that my feature works
- ~I haven't checked if new and existing unit tests pass locally with my changes~
